plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1512,22 +1469,21 @@ public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandl
         Map<String, String> physicalColumnNameMapping = columns.stream()
                 .collect(toImmutableMap(DeltaLakeColumnMetadata::getName, DeltaLakeColumnMetadata::getPhysicalName));
 
-        Map<String, Object> columnTypes = filterKeys(getColumnTypes(metadataEntry), name -> !name.equalsIgnoreCase(dropColumnName));
-        Map<String, String> columnComments = filterKeys(getColumnComments(metadataEntry), name -> !name.equalsIgnoreCase(dropColumnName));
-        Map<String, Boolean> columnsNullability = filterKeys(getColumnsNullability(metadataEntry), name -> !name.equalsIgnoreCase(dropColumnName));
-        Map<String, Map<String, Object>> columnMetadata = filterKeys(getColumnsMetadata(metadataEntry), name -> !name.equalsIgnoreCase(dropColumnName));
+        DeltaLakeTable deltaTable = DeltaLakeTable.builder(metadataEntry, protocolEntry)
+                .removeColumn(dropColumnName)
+                .build();
+        if (deltaTable.columns().size() == partitionColumns.size()) {
```

Comment:
follow-up: This check should be probably be made in `DeltaLakeTable` class when the `DeltaLakeTable` class knows about partitions.

Leaving the partitions aside, we shouldn't be able to remove ALL the columns from a table - even though we check already in `DropColumnTask` about this specific situation.